src/WebJobs.Script.WebHost/Swagger/SwaggerDocumentManager.cs

```diff
@@ -0,0 +1,168 @@
+                    {
+                        InputParameters = inputParameters,
+                        OperationId = $"{endpoint}/{httpOperation}"
+                    };
+                    endpointsOperationData.Add(httpOperation, httpOperationInfo);
+                }
+            }
+            return apiEndpoints;
+        }
+
+        private static string[] GetHttpOperations(IHttpRoute httpRoute)
+        {
+            HttpMethodConstraint httpMethodConstraint = null;
+            httpRoute.Constraints.TryGetValue(ScriptConstants.HttpMethodConstraintName, out httpMethodConstraint);
+            if (httpMethodConstraint != null && httpMethodConstraint.AllowedMethods != null)
+            {
+                return httpMethodConstraint.AllowedMethods.Select(p => p.Method).ToArray<string>();
+            }
+            return ScriptConstants.HttpMethods.ToArray();
+        }
+
+        private static Collection<HttpOperationParameterInfo> GetInputParameters(IHttpRoute httpRoute)
+        {
+            var parameters = new Collection<HttpOperationParameterInfo>();
+            var pathParameters = GetAllPathParameters(httpRoute.RouteTemplate);
+            foreach (var constraint in httpRoute.Constraints)
+            {
+                var httpConstraint = constraint.Value as IHttpRouteConstraint;
+                SwaggerDataType? swaggerDataType = httpConstraint.ToSwaggerDataType();
+                if (swaggerDataType != null && httpConstraint.GetType() != typeof(HttpMethodConstraint))
+                {
+                    var parameter = new HttpOperationParameterInfo()
+                    {
+                        DataType = swaggerDataType.ToString().ToLowerInvariant(),
+                        Name = constraint.Key
+                    };
+                    parameters.Add(parameter);
+                    pathParameters.Remove(constraint.Key);
+                }
+            }
+
+            foreach (var pathParameter in pathParameters)
+            {
+                var parameter = new HttpOperationParameterInfo()
+                {
+                    DataType = SwaggerDataType.String.ToString().ToLowerInvariant(),
+                    Name = pathParameter
+                };
+                parameters.Add(parameter);
+            }
+            return parameters;
+        }
+
+        private static Collection<string> GetAllPathParameters(string routeTemplate)
+        {
+            var pathSegments = new Collection<string>();
+            foreach (var pathSegment in routeTemplate.Split('/'))
+            {
+                if (pathSegment[0] == '{' && pathSegment[pathSegment.Length - 1] == '}')
+                {
+                    pathSegments.Add(pathSegment.Substring(1, pathSegment.Length - 2));
+                }
+            }
+            return pathSegments;
+        }
+
+        public async Task<bool> DeleteSwaggerDocumentAsync()
+        {
+            bool deleted = false;
```

Comment:
Simplify the logic here and update the `FileUtility` to return a `Task<bool>` . This will avoid the double `File.Exists` check and keep things async through.